File: SU_YEON/src/WEEK4/b_1343.java
package WEEK4;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class b_1343 {
    public static void main(String[] args) throws IOException {
        BufferedReader br =  new BufferedReader(new InputStreamReader(System.in));

        String board = br.readLine();

        board = board.replace("XXXX","AAAA");
        board = board.replace("XX", "BB");

        if(board.contains("X")){
            System.out.println("-1");
        }else
            System.out.println(board);
    }
}

File: min_hee/week4_greedy/b1343.py
# 백준 1343 폴리오미노

poly = input().rstrip()

# 1. AAAA로 replace
# 2. BB로 replace
# 3. X가 남아있으면 -1 출력

poly = poly.replace("XXXX", "AAAA")
poly = poly.replace("XX", "BB")

print(poly if poly.count("X")==0 else -1)